### PR TITLE
Added a filter for users without Admin

### PR DESCRIPTION
**Pull request by itsmetoni**

### CODE REVIEW

File: auth-server/src/main/java/com/antonio/authserver/controller/UserController.java [hunk 1]
```diff
@@ -1,91 +1,87 @@
...
-	@PutMapping("/{realmName}/{username}")
-	public ResponseEntity<ResponseMessage> updateUserByUsername(@PathVariable String realmName,@PathVariable String username,
-			@RequestBody AppUserDto appUserDto) {
-		userService.updateUserByUsername(realmName,username, appUserDto);
-		final ResponseMessage responseMessage = new ResponseMessage("User successfully updated");
-		return ResponseEntity.ok().body(responseMessage);
-	}
+    @PostMapping("/{realmName}/{username}/addRole")
+    public void addRoleToUser(@PathVariable String realmName, @PathVariable String username, @RequestBody String roleName) {
+        Role newRole = roleService.findRoleByNameDaoAndRealmName(roleName, realmName);
```
Comment (radasilviu) [inline auth-server/src/main/java/com/antonio/authserver/controller/UserController.java:66]: nu faci aici find, ci in service

File: auth-server/src/main/java/com/antonio/authserver/controller/UserController.java [hunk 2]
```diff
@@ -1,91 +1,87 @@
...
+        userService.deleteUser(realmName, username);
+    }
 
-	@DeleteMapping("/{realmName}/{username}")
-	public void deleteUser(@PathVariable String realmName,@PathVariable String username) {
-		userService.deleteUser(realmName,username);
-	}
+    @DeleteMapping("/{realmName}/{username}/removeRole")
+    public void removeRoleFromUser(@PathVariable String realmName, @PathVariable String username, @RequestBody String roleName) {
+        Role newRole = roleService.findRoleByNameDaoAndRealmName(roleName, realmName);
```
Comment (radasilviu) [inline auth-server/src/main/java/com/antonio/authserver/controller/UserController.java:77]: same, nu aici faci find

File: auth-server/src/main/java/com/antonio/authserver/service/UserService.java [hunk 3]
```diff
@@ -99,43 +98,49 @@ public AppUserDto update(AppUserDto appUserDto) {
...
         appUserRepository.save(appUser);
     }
 
-    public AppUser addRole(String realmName,String username, Role role) throws CustomException {
-        AppUser appUser = appUserRepository.findByUsernameAndRealmName(username,realmName).orElseThrow(() -> new CustomException(
+    public List<AppUserDto> getUsersWithoutAdmins(String realmName) {
+        Role amdinRole = roleRepository.findByName("ROLE_ADMIN").orElseThrow(() -> new CustomException("Role not found", HttpStatus.BAD_REQUEST));
+        List<AppUser> users = appUserRepository.findAll()
+                .stream()
+                .filter(u -> !u.getRoles().contains(amdinRole))
```
Comment (radasilviu) [inline auth-server/src/main/java/com/antonio/authserver/service/UserService.java:120]: nu mai fa read pentru admin, ci cand faci filter, fa direct dupa role name "ROLE_ADMIN"

Comment (itsmetoni) [reply auth-server/src/main/java/com/antonio/authserver/service/UserService.java:120]: Pai in filter am nevoie de obiect, nu pot pune "contains" dupa nume

**Review by radasilviu — APPROVED** [review]

**Pull request merged**